lib/Echidna/Output/Corpus.hs

```diff
@@ -26,3 +31,35 @@ loadTxs dir = do
   putStrLn ("Loaded " ++ show (length txSeqs) ++ " transaction sequences from " ++ dir)
   pure txSeqs
   where readCall f = decodeStrict <$> BS.readFile f
+
+-- save to corpus in the background while tests are running
+-- returns a channel that gets sent () when the process is done
+runCorpusSaver :: Env -> IO (Chan ())
+runCorpusSaver env = case env.cfg.campaignConf.corpusDir of
+  Nothing -> do
+    finishChan <- newChan
+    writeChan finishChan ()
+    pure finishChan
+  Just dir -> do
+    -- we want to dupChan *before* forking so we don't miss any events
+    chan <- dupChan env.eventQueue
+    finishChan <- newChan
+    void $ forkFinally (loop dir chan nworkers) (const $ writeChan finishChan ())
+    pure finishChan
```

Comment:
we can move `finishChan <- newChan` and `pure finishChan` out of the case so its not duplicated